server/controllers/v1/post.ts

```diff
@@ -105,6 +106,74 @@ export default class PostController {
     };
   }
 
+  async getPostsByQuery(
+    userID: string,
+    query: string,
+    limit: number,
+    offset: number
+  ): Promise<{
+    status: number;
+    data: { result?: PostUserPreview[]; count: number; total: number };
+  }> {
+    await db.sequelize.query(`DROP TABLE IF EXISTS "PostSearches";`);
+    await db.sequelize.query(
+      `CREATE TEMP TABLE "PostSearches" AS 
+      SELECT *, (
+        setweight(to_tsvector(coalesce("title",'')), 'A') || 
+        setweight(to_tsvector(coalesce("location",'')), 'C') || 
+        setweight(to_tsvector(coalesce("body",'')), 'D')
+      ) as "tsvector" 
+      FROM "Posts";`
+    );
+    const data = await Promise.all([
+      db.sequelize.query(
+        `SELECT 
+          "PostSearch"."id", 
+          "PostSearch"."body", 
+          "PostSearch"."title", 
+          "PostSearch"."createdAt", 
+          "PostSearch"."thumbnail", 
+          (SELECT COUNT(*) FROM "UserPostLikes" AS "Likes" 
+            WHERE "Likes"."postID" = "PostSearch"."id") AS "likeCount", 
+          (SELECT COUNT(*) FROM "UserPostLikes" AS "Likes" 
+            WHERE "Likes"."postID" = "PostSearch"."id" AND "Likes"."userID" = $1) AS "doesUserLike", 
+          json_build_object(
+            'firstName', "User"."firstName", 
+            'lastName', "User"."lastName",
+            'id', "User"."id"
+          ) AS "User",
+          ts_rank_cd("tsvector", "query", 1|4) AS "rank" 
+        FROM "PostSearches" AS "PostSearch" 
+        LEFT OUTER JOIN "Users" AS "User" ON "PostSearch"."UserId" = "User"."id" 
+        CROSS JOIN to_tsquery($2) AS "query" 
+        WHERE "query" @@ "tsvector" 
+        ORDER BY "rank" DESC 
+        LIMIT $3 
+        OFFSET $4;`,
+        {
+          bind: [userID, query, limit, offset],
+          type: QueryTypes.SELECT,
+        }
+      ) as PostUserPreview[],
+      this.postsRepo.count(),
+    ]);
+
+    return {
+      status: data[0].length > 0 ? 200 : 204,
+      data: {
+        result: data[0].map((p) => {
+          // Must case to `any` as dataValues is not typed at the moment.
+          // Context: https://github.com/RobinBuschmann/sequelize-typescript/issues/760
+          // p.likeCount = (p as any).dataValues.likeCount;
+          // p.doesUserLike = (p as any).dataValues.doesUserLike == 1;
+          return p;
+        }),
```

Comment:
no-op map call???

Reply:
Wasn't sure if it was needed or not, but I guess now I know lol

Also looks like those properties are still present in the response when looking at the dev console